everyvoice/tests/test_preprocessing.py

```diff
@@ -688,6 +688,191 @@ def test_no_speaker(self):
+            tmpdir = Path(tmpdir)
+            fp_config, lj_filelist, _, _, _ = self.get_simple_config(tmpdir)
+
+            # Create filelist with mix of existing and missing files (to avoid SystemExit)
+            wav_filelist = tmpdir / "wav-metadata.psv"
+            with open(wav_filelist, mode="w", encoding="utf8") as f:
+                print("basename|raw_text|characters|speaker|language", file=f)
+                print("LJ050-0269|existing file|existing file|speaker1|en", file=f)
+                print("missing.wav|test text|test text|speaker1|en", file=f)
+
+            fp_config.preprocessing.source_data[0].filelist = wav_filelist
+            fp_config.preprocessing.source_data[0].data_dir = (
+                self.data_dir / "lj" / "wavs"
+            )
+
+            with (
+                capture_stdout(),
+                capture_stderr(),
+                mute_logger("everyvoice.preprocessor"),
+            ):
+                Preprocessor(fp_config).preprocess(
+                    output_path=lj_filelist, cpus=1, to_process=("audio",)
+                )
+
+            # Check missing_files.txt created correctly
+            missing_files_path = tmpdir / "preprocessed" / "missing_files.txt"
+            self.assertTrue(missing_files_path.exists())
+
+            with open(missing_files_path, "r", encoding="utf8") as f:
+                content = f.read()
+            self.assertIn("missing.wav", content)
+
+    def test_empty_missing_files_list_report(self):
+        """Test report method when no missing files exist"""
+        preprocessor = Preprocessor(FeaturePredictionConfig(contact=self.contact))
+
+        # Empty missing files list (default state)
+        self.assertEqual(len(preprocessor.missing_files_list), 0)
+
+        report = preprocessor.report()
+
+        # Should not contain missing files section
+        self.assertNotIn("Missing Audio Files", report)
+
+    def test_no_audio_files_warning(self):
+        """Test that warning is shown when no audio files are processed"""
+        preprocessor = Preprocessor(FeaturePredictionConfig(contact=self.contact))
+
+        # Duration counter is already 0 by default, no need to set it
+        # Verify it's 0
+        self.assertEqual(preprocessor.counters.value("duration"), 0)
+
+        # Create a simple report scenario that triggers the warning
+        with tempfile.TemporaryDirectory(prefix="test_warning", dir=".") as tmpdir:
+            tmpdir = Path(tmpdir)
+            preprocessor.save_dir = tmpdir
+
+            # Simulate the preprocess method's report generation
+            if "audio" in ("audio",):
+                report = "Here is a report:\n" + preprocessor.report()
+                if not preprocessor.counters.value("duration"):
+                    report += "\n\nWARNING: No audio files were processed."
+
+            self.assertIn("WARNING: No audio files were processed.", report)
```

Comment:
Other tests in this suite are great, but I don't understand this one: it's asserting a message that you add here in the unit test, rather than one generated by the preprocessor. Maybe I am misunderstanding something?